lib/ansible/executor/task_executor.py

```diff
@@ -797,7 +797,7 @@ def _poll_async_result(self, result, templar, task_vars=None):
         # that (with a sleep for "poll" seconds between each retry) until the
         # async time limit is exceeded.
 
-        async_task = Task().load(dict(action='async_status jid=%s' % async_jid, environment=self._task.environment))
+        async_task = Task().load(dict(action='async_status', args={'jid': async_jid}, environment=self._task.environment))
```

Comment:
Not something you changed but caught my attention; we seem to needlessly create a Task instance just to call `staticmethod` here instead of:
```suggestion
        async_task = Task.load(dict(action='async_status', args={'jid': async_jid}, environment=self._task.environment))
```

The same with the cleanup task below.

Reply:
Addressed both.